MatrixSDK/JSONModels/MXJSONModels.m

```diff
@@ -2590,3 +2590,48 @@ - (id)copyWithZone:(NSZone *)zone
 }
 
 @end
+
+#pragma mark - Dehydration
+
+@implementation MXDehydratedDevice
+
++ (id)modelFromJSON:(NSDictionary *)JSONDictionary
+{
+    MXDehydratedDevice *device = [[MXDehydratedDevice alloc] init];
+    if (device)
+    {
+        MXJSONModelSetString(device.deviceId, JSONDictionary[@"device_id"]);
+        MXJSONModelSetString(device.account, JSONDictionary[@"device_data"][@"account"]);
```

Comment:
Extract first the dict from `JSONDictionary[@"device_data"]` with `MXJSONModelSetDictionary` to validate it is a NSDictionary.